modnet/models/vanilla.py

```diff
@@ -1534,3 +1549,46 @@ def validate_model(
 
 def map_validate_model(kwargs):
     return validate_model(**kwargs)
+
+
+def generate_shuffled_and_stratified_val_split(
+    y: np.ndarray, val_fraction: float, classification: bool
+):
+    """
+    Generate train validation split that is shuffled, reproducible and, if classification, stratified.
+    """
+    if classification:
+        if isinstance(y[0][0], list) or isinstance(y[0][0], np.ndarray):
+            ycv = np.argmax(y[0], axis=1)
+        else:
+            ycv = y[0]
+        return train_test_split(
+            range(len(y[0])),
+            test_size=val_fraction,
+            random_state=42,
+            shuffle=True,
+            stratify=ycv,
+        )
+    else:
+        return train_test_split(
+            range(len(y[0])), test_size=val_fraction, random_state=42, shuffle=True
+        )
+
+
+def generate_shuffled_and_stratified_val_data(
+    x: np.ndarray,
+    y: list,
+    val_fraction: float,
+    classification: bool,
+):
+    """
+    Generate train and validation data that is shuffled, reproducible and, if classification, stratified.
+    """
+    train_idx, val_idx = generate_shuffled_and_stratified_val_split(
+        y=np.array(y), val_fraction=val_fraction, classification=classification
+    )
+    return (
+        x[train_idx],
+        [t[train_idx] for t in y],
```

Comment:
This doesn't seem correct ?  `y[train_idx]` should work, right?. It would pick the wrong columns now.

Reply:
The `y` of `MODNetModel.fit()` and of `generate_shuffled_and_stratified_val_data` has the first two dimensions (n_target_groups, n_samples) while the `y` of `generate_shuffled_and_stratified_val_split` that is used elsewhere (`data.df_targets.values`) has the first 2 dimensions (n_samples, n_targets).